cupy/functional/piecewise.py

```diff
@@ -0,0 +1,68 @@
+import cupy
+
+from cupy import core
+
+_piecewise_krnl = core.ElementwiseKernel(
+    'U condlist, raw S funclist, int64 xsiz',
+    'raw I prev, raw T y',
+    '''
+        if (condlist){
+          __syncthreads();
+          if (prev[i % xsiz] < i)
+            prev[i % xsiz] = i;
+          __syncthreads();
+          if (prev[i % xsiz] == i)
+            y[i % xsiz] = funclist[i / xsiz];
+        }
+        ''',
+    'piecewise_kernel'
+)
+
+
+def piecewise(x, condlist, funclist):
+    """Evaluate a piecewise-defined function.
+
+        Args:
+            x (cupy.ndarray): input domain
+            condlist (cupy.ndarray or list):
+                conditions list ( boolean arrays or boolean scalars).
+                Each boolean array/ scalar corresponds to a function
+                in funclist. Length of funclist is equal to that
+                of condlist. If one extra function is given, it is used
+                as otherwise condition
+            funclist (cupy.ndarray or list): list of scalar functions.
+
+        Returns:
+            cupy.ndarray: the result of calling the functions in funclist
+                on portions of x defined by condlist.
+
+        .. warning::
+
+            This function currently doesn't support callable functions,
+            args and kw parameters.
+
+        .. seealso:: :func:`numpy.piecewise`
+        """
+
+    if any(callable(item) for item in funclist):
+        raise NotImplementedError(
+            'Callable functions are not supported currently')
+    if cupy.isscalar(condlist):
+        condlist = cupy.full(shape=x.shape, fill_value=condlist)
+    if not isinstance(condlist[0], (list, cupy.ndarray)) and x.ndim != 0:
```

Comment:
Could you explain why the codition `x.ndim != 0` is needed?
Please add the test for 0-dim input.

Reply:
needed for scalar input

Reply:
Sorry, I noticed now this line is already outdated :bow: